### ISSUE TITLE
Damage of ProfitTrailerData.json

### ISSUE THREAD

**Issue by jokaorgua** [taniman/profit-trailer#1919]
NO NEW FEATURES! Use [PT Enahancements](https://github.com/taniman/profit-trailer-enhancements) for Feature Requests.

## ProfitTrailer info
Version: 2.0.10

 
Exchange:
- [ ] Bittrex
- [x] Binance
- [ ] Poloniex

 
Market:
- [x] BTC
- [ ] ETH
- [ ] USDT
- [ ] BNB
- [ ] XMR

 
Operating System:
- [ ] Windows (please specify version)
- [ ] Ubuntu (please specify version)
- [x] Other linux distro (please specify) CentOS
- [ ] Mac OS

## Expected Behavior
No errors while saving the data file or automatic restore from backup

## Current Behavior
2018-06-05 12:06 +00:00: 2018-06-05 12:06:33 ERROR Util - Both save files were damaged
2018-06-05 12:06 +00:00: 2018-06-05 12:06:33 ERROR Util - If you have a backup somewhere please use that or just delete ProfitTrailerData.json

## Troubleshooting
Manually restored from backup dir

## Possible Solution
Automatic restore

## Steps to Reproduce
pm2 restart profit-trailer

**Comment by taniman** [comment]
Sorry we only do automatic restore from the profitrailerdata.json.backup. Going into the backup folder to restore is a bit out of scope and is not actually a bug.

I have been running months without ever getting a corrupted save file. So something happened on your machine that caused this corruption.

Maybe once we start storing this data in a database of sorts this will be less likely to happen.  

tl;dr: Reading the backup folder is not something I want to do as a bug fix

**Comment by jokaorgua** [comment]
so find other way of saving or using your data. bot should NOT corrupt the data and your backup folder is just big HACK

**Comment by jokaorgua** [comment]
BTW. pm2 with restart command sends SIGINT to application. So will be enough to finish correct file save in SIGINT handler.

**Comment by taniman** [comment]
I also use PM2 on my machines. Never had the problem. but I will check this sigint .
